File: rules/python/no-request-timeout.py
# ruleid: no-request-timeout
resp = requests.get(url)

# ruleid: no-request-timeout
requests.post(endpoint, json=payload)

# ok: no-request-timeout
resp = requests.get(url, timeout=10)

# ok: no-request-timeout
client = httpx.Client(timeout=5)

# ok: no-request-timeout
resp = requests.get(url, timeout=(3.05, 27))

File: rules/javascript/no-request-timeout.js
const request = require("request");

// ruleid: no-request-timeout
request.get(url);

// ruleid: no-request-timeout
request.post(endpoint, { json: payload });

// ruleid: no-request-timeout
request.put(api, body, { json: true });

// ok: no-request-timeout
request.get(url, { timeout: 10000 });

// ok: no-request-timeout
request.post(endpoint, { json: payload, timeout: 5000 });

// ok: no-request-timeout
axios.get(url, { timeout: 5000 });